src/main/java/io/github/pmckeown/dependencytrack/upload/UploadBomMojo.java

```diff
@@ -78,10 +81,11 @@ public UploadBomMojo(UploadBomAction uploadBomAction, MetricsAction metricsActio
 
     @Override
     public void performAction() throws MojoExecutionException, MojoFailureException {
+        enrichCommonConfig();
         logger.info("Update Project Parent : %s", updateParent);
 
         try {
-            if (!uploadBomAction.upload(getBomLocation(), isLatest, projectTags)) {
+            if (!uploadBomAction.upload()) {
                 handleFailure("Bom upload failed");
             }
             Project project = projectAction.getProject(projectName, projectVersion);
```

Comment:
@thomashucke  - with this change, there is now a bug if the `<updateParent>` option is set to true and the UUID is set.

On line 98, the project parent is fetched by name and version but this needs to be updated to use parentUuid as a priority if its been set.

This is the output of a local build when testing.

Can you add some code to handle this?  I can't merge this until this is fixed.

<img width="1420" alt="image" src="https://github.com/user-attachments/assets/b3629101-876a-4eb0-9210-2cbfe9f53938">

Reply:
@pmckeown - it was a bit more tricky as expected but I think it should be ok now.
Added also some documentation to consider specialties in a mutli-module setup.

Reply:
This is fixed now.  Nice one.

![image](https://github.com/user-attachments/assets/4e79217d-0ec1-4610-90bd-54a8cbaf0359)